packages/documentation/src/content/docs/integration/deployment/helm-k8s.mdx

```diff
@@ -3,49 +3,823 @@ title: Helm & Kubernetes
+```
+
+Note that this won't delete [Persistent Volume Claims (PVC)](https://kubernetes.io/docs/concepts/storage/persistent-volumes/) created by the PostgreSQL and Redis deployments. If you want to delete them as well:
+
+```bash
+kubectl delete pvc -l app.kubernetes.io/instance=rafiki
+```
+
+## Troubleshooting
+
+### Checking pod logs
+
+If a component isn't working correctly, you can check its logs:
+
+```bash
+# List all pods
+kubectl get pods
+
+# Check logs for a specific pod
+kubectl logs pod/rafiki-auth-server-0
+```
+
+### Checking resources and logs
+
+```bash
+# List pods and their status
+kubectl get pods
+
+# Check logs for a specific pod
+kubectl logs pod/rafiki-auth-server-0
+
+# Get details about a pod
+kubectl describe pod/rafiki-auth-server-0
+
+# Check services and their endpoints
+kubectl get services
+
+# Check Persistent Volume Claims
+kubectl get pvc
+
+# Check Ingress resources
+kubectl get ingress
+```
+
+## Common Issues
+
+### Database connection errors
+
+1. Check if PostgreSQL pods are running: `kubectl get pods -l app.kubernetes.io/name=postgresql`
+2. Check PostgreSQL logs: `kubectl logs pod/rafiki-postgresql-0`
+3. Verify that the database passwords match those in your values.yaml
+
+### Tigerbeetle initialization failures
+
+1.  Check Tigerbeetle logs: `kubectl logs pod/tigerbeetle-0`
+2.  Ensure that the PVC for Tigerbeetle has been created correctly `kubectl get pvc -l app.kubernetes.io/name=tigerbeetle`
+3.  Verify that the cluster ID is consistent across all components
+
+### Ingress issues
+
+1.  Verify NGINX Ingress Controller is running: `kubectl get pods -n ingress-nginx`
+2.  Check if your DNS records are correctly pointing to the ingress controller's external IP
+3.  Check the Ingress resource: `kubectl get ingress`
+4.  Check Ingress controller logs: `kubectl logs -n ingress-nginx deploy/nginx-ingress-ingress-nginx-controller`
+5.  Verify that TLS secrets exist if HTTPS is enabled: `kubectl get secrets`
+
+### TLS certificate problems
+
+1.  If using cert-manager, check if certificates are properly issued: `kubectl get certificates`
+2.  Check certificate status: `kubectl describe certificate [certificate-name]`
+3.  Check cert-manager logs: `kubectl logs -n cert-manager deploy/cert-manager`
+
+### Service unavailable
+
+1.  Check if services are running: `kubectl get services`
+2.  Verify pod health: `kubectl describe pod [pod-name]`
+3.  Check for resource constraints: `kubectl top pods`
+
+### Connectivity between components
+
+1.  Ensure all required services are running: `kubectl get services`
+2.  Verify service endpoints: `kubectl get endpoints`
+3.  Test connectivity between pods using temporary debugging pods:
+    ```bash
+    kubectl run -it --rm debug --image=busybox -- sh
+    # Inside the pod
+    wget -q -O- http://rafiki-auth-server:3000/health
+    ```
+
+## Security considerations
+
+When deploying Rafiki in production, consider the following security practices:
+
```

Comment:
Ensure the list in this section is parallel

- Use secure passwords
- Enable TLS
- Implement network policies
- Use RBAC
- Use secrets management
- Perform regular updates